app/components/sidebars/main/channels_list/filtered_list/filtered_list.js

```diff
@@ -82,14 +86,13 @@ class FilteredList extends Component {
         return !deepEqual(this.props, nextProps, {strict: true}) || !deepEqual(this.state, nextState, {strict: true});
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.term !== nextProps.term) {
-            const {actions, currentTeam} = this.props;
-            const {term} = nextProps;
+    static getDerivedStateFromProps(props, state) {
+        if (state.term !== props.term) {
+            const {actions, currentTeam} = state;
+            const {term} = props;
             const {searchChannels, searchProfiles} = actions;
-            const dataSource = this.buildData(this.props, term);
+            const dataSource = this.buildData(state, term);
 
-            this.setState({dataSource, term});
             clearTimeout(this.searchTimeoutId);
 
             this.searchTimeoutId = setTimeout(() => {
```

Comment:
Can you move this out into `componentDidUpdate`? `getDerivedStateFromProps` shouldn't have any side effects and should only be used to update the state